atlp-rwanda/eagles-ec-be
# Ft profile page setting #187419175

**What does this PR do?**

This PR implements profile page setting by using generic and user specific data/info.

**Description of Task to be completed?**

Implement the feature to allow user to create a profile
Can be able to retrieve and see their profile once they log in 
Can be able to update their profile once they log in
Can not be able to update their email.
Updating full name, and country is compulsory

**How should this be manually tested?**

Clone the repository.
Checkout to the branch ft-profilePage-setting-#1874419175.
Run npm install to install dependencies.
Run npm run dev to start the development server.
Open a web browser and navigate to the /docs endpoint (e.g., http://localhost:PORT/docs).
Verify that the Swagger UI loads correctly.
First authorize by logging in any user and use the token provided
Explore the endpoint **GET on /api/v1/users/profile** and  **PATCH on /api/v1/users/profile** user should be able to retrieve his or her profile and able to update their existing data in their profile.
What are the relevant pivotal tracker stories?

[Finished #187441917]

**Screenshort**

![Screenshot (6)](https://github.com/atlp-rwanda/eagles-ec-be/assets/105001832/5ccb058d-747b-4b58-9be1-8ef4036df261)
![Screenshot (7)](https://github.com/atlp-rwanda/eagles-ec-be/assets/105001832/8092c21c-285d-4d59-9556-3de47472dbce)
